libmachine/drivers/drivers.go

```diff
@@ -107,3 +108,22 @@ func MustBeRunning(d Driver) error {
 
 	return nil
 }
+
+// DriverUserdataFlag returns true if the driver is detected to have a userdata flag.
+func DriverUserdataFlag(d Driver) string {
+	for _, opt := range d.GetCreateFlags() {
+		if NameIsUserData(opt.String()) {
+			return opt.String()
+		}
+	}
+
+	return ""
+}
+
+// NameIsUserData returns true if the given flag is a userdata flag
+func NameIsUserData(name string) bool {
+	return strings.Contains(name, "user-data") ||
+		strings.Contains(name, "userdata") ||
+		strings.Contains(name, "custom-data") ||
+		strings.Contains(name, "cloud-config")
```

Comment:
Where do these come from ? If at random we should lower all of them to catch camel case.

Reply:
These are flags from the different drivers that support user data. The flags are all lowercase.